logstash-input-webfetcher/src/main/java/eu/wajja/web/fetcher/controller/URLController.java

```diff
@@ -121,4 +129,19 @@ private void closeConnection(HttpURLConnection httpURLConnection) {
 		return new byte[0];
 	}
 
-}
\ No newline at end of file
+    public URL createUrl(String currentUrl){

```

Comment:
if url is invalid, this will create nullpointer on line 47
And do we really need this ?

Reply:
It is what mockito suggests:
https://github.com/mockito/mockito/wiki/Mocking-Object-Creation#pattern-1---using-one-line-methods-for-object-creation
Caught the exception in the later in the chain, to avoid the null pointer, but still trigger the exception.